module/entityAssessors/helper.js

```diff
@@ -193,17 +193,30 @@ module.exports = class entityAssessorHelper {
                     name: { $in: Object.values(entityTypeQueryList) }
                 }, { name: 1 }).lean();
 
-                let userIds = assessorData.map(assessor => assessor.userId);
+                let userExternalIds = assessorData.map(assessor => assessor.externalId);
+                let parentExternalIds = assessorData.map(assessor => assessor.parentId);
+
+                userExternalIds.push(...parentExternalIds);
+
+                let userIds = await Promise.all(userExternalIds.map(async(loginId) => {
+                    return this.getInternalUserIdByExternalId(token, loginId)
+                }))
 
-                let entityAssessorDocument = await database.models.entityAssessors.find({ userId: { $in: userIds } }, { entities: 1, userId: 1 }).lean();
+                let userIdByExternalId={};
+
+                userIds.forEach(userId=>{
+                    if(userId) userIdByExternalId[Object.keys(userId)[0]] = Object.values(userId)[0].userId;
+                })
+
+                let entityAssessorDocument = await database.models.entityAssessors.find({ userId: { $in: Object.values(userIdByExternalId) } }, { entities: 1, userId: 1 }).lean();
```

Comment:
Why no filter of the solution and program Id here?

Also, above userIds are combination of parentid and userid, so no point loading parentId as entity assessor documents, since they will not be updated.

@ajithtunerlabs

Reply:
in the old structure, we are getting school fields from school assessor but now we don't have any fields so we are constructing all the field from csv itself ... so i removed `fieldsWithOutSchool `

Reply:
addressed